src/util/transition/ngbTransition.ts

```diff
@@ -74,7 +74,8 @@ export const ngbRunTransition =
           // because 'transitionend' event might not be fired in some browsers, if the transitioning
           // element becomes invisible (ex. when scrolling, making browser tab inactive, etc.). The timer
           // guarantees, that we'll release the DOM element and complete 'ngbRunTransition'.
-          const transitionEnd$ = fromEvent(element, 'transitionend').pipe(takeUntil(stop$));
+          const transitionEnd$ =
+              fromEvent(element, 'transitionend').pipe(takeUntil(stop$), filter(({target}) => target === element));
```

Comment:
Same as my previous comment. We should comment here that filter is to discard all _inner_ transition.

Question: what if the _inner_ transition is **longer** than the outer one ? Generally speaking we should wait for the longer one ?

Reply:
Will add a comment, thanks. Just forgot to do this.

> we should wait for the longer one ?

No, I don't think so. We have no idea what would be inside, so we shouldn't wait. If they really need to wait for some transition inside, that can just make modal transition longer